ibis/backends/duckdb/__init__.py

```diff
@@ -67,6 +67,51 @@ def do_connect(
         )
         self._meta = sa.MetaData(bind=self.con)
 
+    def register(
+        self,
+        file_name: str | Path,
+        table_name: str | None = None,
+    ) -> None:
+        """Register an external file (csv or parquet) as a table in the current
+        connection database
+
+        Parameters
+        ----------
+        file_name
+            Name of the parquet or CSV file
+        table_name
+            Name for the created table.  Defaults to filename if not given
+        """
+        file_name = Path(file_name)
+        suffix = "".join(file_name.suffixes).strip(".")  # handles .csv.gz
+        if file_name.parts[0].endswith(":"):
+            prefix, *fname = file_name.parts
+        else:
+            prefix = "file:"
+            fname = file_name.parts
+
+        file_name = Path(*fname).absolute()
+
+        # Use prefix for file_type.  If omitted, infer from file extension
+        file_type = prefix.strip(":") if prefix != "file:" else suffix
+        table_name = table_name or file_name.stem.replace("-", "_")
+        if file_type == "parquet":
+            view = f"""
+            CREATE VIEW {table_name} as SELECT * from
+            read_parquet('{file_name}')
+            """
+        elif file_type.startswith("csv"):
```

Comment:
What do you think about ad(a|o)pting this code https://github.com/blaze/odo/blob/master/odo/regex.py so we can avoid all the branching?

We can probably make use of this in `ibis.connect` as well.

Reply:
`blaze` really is the phoenix.